### DISCUSSION TITLE
Beelance Improvements

### DISCUSSION THREAD

**Discussion by TemperedFool** [NillerMedDild/Enigmatica6#1695] [General]
Three primary suggestions stemming from a Discord discussion:

1. Resoruceful Bee Hive/Apiary costs scale very fast, especially in the context of Normal mode. I think we should be encouraging players to push through from the basic hives to at least the T1 Apiary, to encourage a reduction in the number entities flying around while still being able to scale up production. I think the production of each tier is probably fine, but the 4:1 scaling as you tier up is brutal and encourages going wide rather than scaling up. I think I would flatten that to 1:1 at least through the T1 Apiary, and would make the ratio for further apiary scaling at most 2:1 (although my preference I think is 1:1 the whole way). I'd also remove the nether star requirement for the T1 Apiary; @MuteTiefling suggested netherite, which seems reasonable.
2. Encourage breeding & make it easier to find all of the native bees by reducing the naturally spawning bees to a smaller number. Model this similar to Forestry, but with bees that are productive - Clay, Coal, Gravel, I-C, maybe a Oak Bee. Keep the number of naturally-spawning bees in the 6-8 range, and require players to breed the rest. The breeding mechanic for Resoruceful Bees isn't punishing, so this seems like we could make it work.
3. Don't have many/most of the combs directly produce things like metals and gems - shift them to produce chunks or other intermediaries that still require processing. I would personally just lean on EE chunks, and probably focus on metal/gem bees for this. This way players can't just skip ore processing if they want to be efficient, and are rewarded if they build out an efficient ore processing scheme to go along with their bees.

If we want to move forward with this, it can move to an Issue for tracking purposes.

**Comment by MaxNeedsSnacks** [comment]
Another idea I've had was to remove the 100% success rate from some of the crossbreeds. The element of chance introduced a bit of a challenge back in the days of 1.12, and while with Forestry, it was sometimes quite painful to get the bees you wanted reliably (as even breeding Forest/Forest could give you a non-Forest bee because of recessive traits and Mendelian inheritance, for example), I feel that with Resourceful Bees, the pain is MUCH much more manageable, and a more intriguing gate than just leaving it up to time once.

**Reply by TemperedFool** [reply]
For the breeding part, I think this breaks out into three different things to plan:

- What bees spawn in the wild?
- What does the breeding tree look like?
- What should the breeding costs be?

The current "many of whatever each parent bee produces" costs probably aren't ideal, but are especially punishing if there is a chance of a crossbreed attempt failing.

**Reply by NielsPilgaard** [reply]
We need to bring the cost way down if breeding becomes the intended way to obtain them. Maybe 1/8 of current costs.
Edit: I quite like the idea of decreasing the chance to say, 50-95% depending on the bee tier, as long as it doesn't feel too punishing to have it fail multiple times.

**Reply by TemperedFool** [reply]
Reducing the breeding cost + making tiering up to the Apiary easier for its breeder will make it way less punishing.

**Comment by merlerm** [comment]
I like the idea of limiting the number of the ones available in the wild significantly to 4/5 variants (plus maybe a nether/end -undergarden?- one) and making the higher tier bees only available via crossbreed or even mutation, which is a fun mechanic the mod itself doesn't use much. Reducing the amount of randomness from worldgen is really good, since having to travel thousands of blocks to *maybe* find a bee you want is frustrating at best, and breeding them can be a fun mechanic, especially since RB already covers the mechanic fairly in depth. Having the more expensive bees only available via tier up and with some work behind them is also pretty fair, and it would allow us to maybe introduce even weirder bee variations on our part

**Comment by TemperedFool** [comment]
PR #1705 should cover 1. Now to formally chart everything for 2 and 3!

**Reply by NielsPilgaard** [reply]
I'm thinking those weren't meant to be links :P

**Reply by TemperedFool** [reply]
1/2/3 weren't, but it's kinda amusing to have them accidentally link to random issues...

**Comment by TemperedFool** [comment]
What do people think about 3? Restructuring the breeding tree for 2 is a pretty big project, but if there is consensus that moving to outputting chunks for those bees who's products have them is a good plan, that seems like it can be done pretty quickly.

**Reply by MaxNeedsSnacks** [reply]
I honestly wouldn't want the bees to output chunks directly, as especially at higher tiers their outputs get... kinda ridiculous, maybe tiny chunks would actually be more fitting here (those are already part of newer versions of EE, as well)

**Reply by TemperedFool** [reply]
Yeah, I was thinking little chunks rather than big chonks. Allows bees to be your ore gen, but rewards the silly 4/5x setups people might otherwise build. 

Means that for general ore gen bees, these are competing with Astral(mineralis)/Botania(orechid)/Occultism(djinn miner that I know nothing about)/Create monstrosities and other quarries.

**Comment by TemperedFool** [comment]
Capturing the biome/breeding suggestions Max had here so they don't get lost. Starting to chart this out a bit is the next step for Item 2:

```
So after a bit of work, I think this might be a good working draft for low-tier bees (aka ones that spawn in the wild):

## Tier 1 Bees (appear commonly in-world):
(Vanilla)
- Common (Vanilla bee, produces "regular" Honeycombs)

(Biome)
- Forest (appears in Forests, produces regular and Silky Combs)
- Winter (appears in Cold Biomes, produces regular and Snowy Combs)
- Desert (appears in Desert Biomes, produces regular and Dry Combs)
- Tropical (appears in Jungles, produces Silky and rarely Mossy Combs)
- Marsh (appears in Swamps and Marshlands, produces Watery and rarely Slimy Combs)

(Elemental / Dimensional)
- Igneous (spawns underground and near lava lakes, produces Molten(?) Combs)
- Aqueous (spawns near rivers and beaches, produces Watery Combs)

- Infernal (Nether equivalent to "Common", produces Hellish Combs)
- Alien (End equivalent to "Common", produces Otherworldly Combs)

For reference, the Dimensional Bees would have their own breeding tree which will not be expanded on for now.

## Tier 2 Bees (appear rarely in-world, can be bred from Tier 1 otherwise):
- Wooden (may be different Bees like Oak, Spruce, ...) = Common + [Biome Bee] (can be found naturally, produces Wooden Combs)
- Earthen = Common + [Biome Bee] (can be found in Biomes as well, produces Dirty Combs)
- Stone = Igneous + Aqueous (seen in Caves, produces Rocky Combs and may mutate nearby stones into random igneous ones)
- Slimy = Marsh + Aqueous (can also be found in Swamps together with Marsh Bees, produces Slimy Combs)
- [...] possibly more bees based on Tier 1 ones

## Tier 3 (could possibly be moved into Tier 2)
- Metamorphic / Sedimentary = Stone + [Igneous/Aqueous] (produces Rocky Combs and may mutate stones into metamorphic and sedimentary ones, respectively)
- Gravel / Sand = [Stone/Metamorphic/Sedimentary] + [Igneous or Aqueous] (produce Flinty and Sandy Combs, respectively, may crush nearby Cobble -> Gravel, Gravel -> Sand)
- Mossy = Stone + Tropical (produces Mossy Combs)
```

![](https://cdn.discordapp.com/attachments/730169148294037544/820750505750954004/unknown.png)
![](https://images-ext-1.discordapp.net/external/2fed45PcAIGZjWK-xbqU94Hj3Gp2KIy8T0j0R9aq84o/https/img.someone.ky/chrome_45d1b9c6-beb6-47b7-9cad-ecfb8aa792ba.png)

**Comment by KrAzYGEEK32** [comment]
i really like this complete overhaul of bees, all the 3 points are very exciting! 

i do wanna add some more insights as to why the current bee production isn't beelanced enough though. in a T4 apiary a bee produces 2 comb blocks, thats 18 combs per cycle, if you use 2 of the same bees in the apiary. where as if you use 2 t4 hives and spam 32 of same bees, you get 32 combs per cycle(although the cycle is a little bit longer). which is way more output that a t4 apiary. breeding 32 bees is very easy with a TIAB and an apiary breeder. as T1 apiary is way cheaper now, getting the breeder is also very easy. this encourages people( like me XD) to stay at lower levels and spam more bees, instead of working towards a better apiary.

**Reply by TemperedFool** [reply]
Interesting. I'll admit I only tested at 1:16 rather than 2:32, but the last time I ran an extended production test the T1 Apiary performed similarly to the T4 hive at those population counts. I think I'm going to have to set this up in the latest dev and let it run for a bit, see where things lie at the moment in practice.

**Reply by KrAzYGEEK32** [reply]
my main thought on a solution would be to buff the t4 apiary outputs to 4 from 2 blocks, similarly t3 from 1 to 2, or make the back to their default values i.e. t3 - 4, t4 - 8. anything works, do keep in mind that im talking about a specific case when you need a single type of resource

**Reply by TemperedFool** [reply]
So, I've set up a test setup in the latest dev environment. I will for sure say I don't know if I've optimized the T4 Hive - that many bees in a tiny space seemed to basically just spend their time bumping into each other rather than finding flowers. The setup looks like this:

<img width="1806" alt="image" src="https://user-images.githubusercontent.com/13169307/112208217-7e289c00-8bee-11eb-905e-66b760a24c94.png">

I have only been running it for ~20 minutes or so, and have the following outputs so far:

<img width="562" alt="image" src="https://user-images.githubusercontent.com/13169307/112208682-0149f200-8bef-11eb-8590-c044eaa39fe3.png">

Let me know if you have suggestions for optimizing the hive, as so far it is getting completely wrecked by the T4 apiary, and the T1 is actually more or less keeping pace with it!

**Reply by TemperedFool** [reply]
Just to clarify, this is 32:2 Hive/Apairy population counts, just using iron bees. The collection system for the Hive is using Pedestals:

<img width="1806" alt="image" src="https://user-images.githubusercontent.com/13169307/112208938-50902280-8bef-11eb-8383-34bdecbe35ad.png">

**Reply by TemperedFool** [reply]
This is definitely a case of "run it long enough to get a good sample". I swapped to the dispenser method just in case the Pedestals upgrade was being weird. I will have real numbers later, but when I checked on things before bed last night the hive was ahead of the T1 and T2. I will need to think through this a bit ^_^

**Reply by TemperedFool** [reply]
@KrAzYGEEK32 Ok, so it looks like the T2 Apiary produces roughly the same as the T4 Hive at that 2:32 bee count. My thinking is that boosting the T1 & T2 apiaries to 5 & 10 combs respectively may work out nicely; the T1 apiary is still a SLIGHTLY lower output than the hive, but it's close, and with the new apiary costs it's not expensive to go wide if you prefer that to tiering up. I'm personally more apt to tier up and not min-max, so hearing the thoughts of a min-maxer would help ^_^

**Reply by TemperedFool** [reply]
People love nerfs, but a better way of dealing with this imbalance may be to cut the number of bees allowed in hives by half. This gives us headroom to either adjust production in apiaries & hives up later, or increase the likelihood that the centrifuge is productive, to put bees in line with other ore generation options...

**Reply by KrAzYGEEK32** [reply]
I agree, the bee count is way too high for t4 hives for example, I'm thinking along the lines of 10 bees for t4 hives and balance the rest hives from there. I feel like even 10 is too high lol. Maybe reduce hive times to soften the blow a bit. I also noticed that chances for most of the outputs is pretty crap XD, like.. why would you make gravel output based on chance?! I was particularly interested in uranium, ands it's 10% per comb, I got 56 ingots from a stack of uranium comb blocks

**Reply by KrAzYGEEK32** [reply]
I know that's not the optimal way to farm uranium from bees but still, that just felt like RNG trolling me XD.

**Reply by KrAzYGEEK32** [reply]
Also, I will take your word for the numbers on hives vs apiaries :D , I didn't extensively test them, i just felt the t4 hives are performing way better than t1 apiaries and pointed it out.

**Reply by Quezler** [reply]
off topic, but: i randomly landed on this discussion and noticed the seamless glass at the top with the "golden" edge on @TemperedFool 's picture(s), would anyone happen to know the name of it? 🤔 (or which mod its from / anything helpful)

**Comment by KrAzYGEEK32** [comment]
one more idea for incentivizing apiaries i had was, if we could remove or increase the cap of 1 unique bee per apiary condition of it. that would make it way more efficient in terms of resource gathering, and you wont need to spam more of them. it could be too OP if we totally remove the cap, but increasing it would be a neat balance, maybe 2 or 4 bees per apiary. we do need a lot of certain resources *cough* uranium * cough* and there are not many renewable ways to generate them. im hoping against hope tat this is possible with configs, let me know if it isnt :D

**Reply by TemperedFool** [reply]
Yeah I kinda like that idea, but I think the configs don't allow it. The thing is that from a resource production POV, the apiary itself is super efficient (producing ~450% more at T1 than a T4 Hive), it just needs to be varied. Build a second apiary, and you can have the same production as one of those materials, and 900% total production efficiency. I know what you mean by needing a ton of a specific material, but I think the allure of the apiary really is "but I can make that much with 4 bees instead of 32, and ALSO make that much of 8 other materials or half as much of 16". That either sounds appealing or it doesn't ^_^

**Comment by ManiacRobertWasTaken** [comment]
What if instead of ingots we change with chunks? Bees in "early-game" is too expensive and in "end-game" is useless.

**Reply by NielsPilgaard** [reply]
I believe that is planned.

**Reply by KrAzYGEEK32** [reply]
but we already have a better way of making chunks using bees, what would be the point of apiaries then?
as block mutations are way better than combs already

**Comment by TemperedFool** [comment]
Work has started on 2 and 3 with PR #2000!
